### PR TITLE
added better test code coverage for did peer

### PR DESCRIPTION
**Pull request by andorsk**
adding more test coverage to did peer. https://github.com/TBD54566975/ssi-sdk/issues/209

### CODE REVIEW

**Comment by andorsk** [comment]
@decentralgabe sorry for taking so long on this. Just haven't had a lot of cycles to spare lately.

File: did/peer.go [hunk 1]
```diff
@@ -258,11 +259,21 @@ func (d DIDPeer) buildVerificationMethod(data, did string) (*VerificationMethod,
...
 	d, ok := did.(DIDPeer)
 	if !ok {
 		return nil, errors.Wrap(util.CastingError, "did:peer")
 	}
 
+	//if !d.IsValid() {
+	//return nil, errors.New("did is not valid")
+	//}
+
+	ds := string(d)
```
Comment (decentralgabe) [inline did/peer.go:272]: can you add a comment here please?

File: did/peer_test.go [hunk 2]
```diff
@@ -12,6 +12,146 @@ func TestDIDPeerValid(t *testing.T) {
...
+		s2, err := d.encodeService(sbe)
+		assert.NoError(tt, err)
+		assert.Equal(tt, res, s2)
+	})
+
+}
+
+func TestPeerResolver(t *testing.T) {
+	bad := "asdf"
+	r := PeerResolver{}
```
Comment (decentralgabe) [inline did/peer_test.go:121]: `var r PeerResolver`

File: did/peer_test.go [hunk 3]
```diff
@@ -79,6 +218,13 @@ func TestPeerMethod2(t *testing.T) {
 	assert.True(t, did.IsValid())
 }
 
+func TestPeerMethod1(t *testing.T) {
+	m1 := PeerMethod1{}
```
Comment (decentralgabe) [inline did/peer_test.go:222]: `var m1 PeerMethod`

**Review by decentralgabe — CHANGES_REQUESTED** [review]

**Comment by decentralgabe** [comment]
thanks for taking the time to add these @andorsk !

File: did/peer.go (hunk 1 above)
Comment (andorsk) [reply did/peer.go]: done. This had to do with padding. I changed the method during the service decoding section.

File: did/peer.go [hunk 4]
```diff
@@ -258,11 +259,20 @@ func (d DIDPeer) buildVerificationMethod(data, did string) (*VerificationMethod,
 // Extract element purpose and decode each key or service.
 // Insert each key or service into the document according to the designated pu
 func (m PeerMethod2) resolve(did DID, _ ResolutionOptions) (*DIDResolutionResult, error) {
+
 	d, ok := did.(DIDPeer)
 	if !ok {
 		return nil, errors.Wrap(util.CastingError, "did:peer")
 	}
 
+	// The '=' at the end is an artifact of the encoding, and will mess up the decoding
```
Comment (decentralgabe) [inline did/peer.go:268]: did you mean to comment this out?

**Review by decentralgabe — APPROVED** [review]

**Comment by andorsk** [comment]
oh. sorry. my mistake. misread the merge as it was going into main. I'll reopen it.

**Pull request merged**